tests/slate/test_assemble_tensors.py

```diff
@@ -306,16 +303,16 @@ def test_diagonal(mass, matrix_mixed_facet):
         assert np.allclose(r, d, rtol=1e-14)
 
     # test matrix built from diagonal for non mass matrix
-    res2 = assemble(DiagonalTensor(Tensor(matrix_mixed_facet))).M.values
-    ref2 = np.concatenate(assemble(matrix_mixed_facet, diagonal=True).dat.data)
+    res2 = assemble(DiagonalTensor(Tensor(matrix_mixed_nofacet))).M.values
+    ref2 = np.concatenate(assemble(matrix_mixed_nofacet, diagonal=True).dat.data)
     for r, d in zip(res2, np.diag(ref2)):
         assert np.allclose(r, d, rtol=1e-14)
 
     # test matrix built from diagonal
     # for a Slate expression on a non mass matrix
-    A = Tensor(matrix_mixed_facet)
+    A = Tensor(matrix_mixed_nofacet)
     res3 = assemble(DiagonalTensor(A+A)).M.values
-    ref3 = np.concatenate(assemble(matrix_mixed_facet+matrix_mixed_facet, diagonal=True).dat.data)
+    ref3 = np.concatenate(assemble(matrix_mixed_nofacet+matrix_mixed_nofacet, diagonal=True).dat.data)
```

Comment:
Is it the case that slate does the wrong thing for `DiagonalTensor(something_with_interior_facet_terms)`? Or is it that the test no longer passes? I think if it is the former should the slate compiler also complain one asks for `DiagonalTensor` of interior facet?

Reply:
TSFC throws the `NonImplementedError`